pkg/service/blob_storage.go

```diff
@@ -0,0 +1,114 @@
+package service
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"mime"
+	"strings"
+	"time"
+
+	"github.com/gabriel-vasile/mimetype"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/instill-ai/pipeline-backend/pkg/recipe"
+	"github.com/instill-ai/pipeline-backend/pkg/resource"
+	"github.com/instill-ai/pipeline-backend/pkg/utils"
+
+	artifactpb "github.com/instill-ai/protogen-go/artifact/artifact/v1alpha"
+	resourcex "github.com/instill-ai/x/resource"
+)
+
+func (s *service) uploadBlobAndGetDownloadURL(ctx context.Context, ns resource.Namespace, data string) (string, error) {
+	mimeType, err := getMimeType(data)
+	if err != nil {
+		return "", fmt.Errorf("get mime type: %w", err)
+	}
+	artifactClient := s.artifactPublicServiceClient
+	requesterUID, _ := resourcex.GetRequesterUIDAndUserUID(ctx)
+
+	vars, err := recipe.GenerateSystemVariables(ctx, recipe.SystemVariables{})
+
+	if err != nil {
+		return "", fmt.Errorf("generate system variables: %w", err)
+	}
+
+	ctx = metadata.NewOutgoingContext(ctx, utils.GetRequestMetadata(vars))
+
+	timestamp := time.Now().Format(time.RFC3339)
+	objectName := fmt.Sprintf("%s-%s%s", requesterUID.String(), timestamp, getFileExtension(mimeType))
+
+	resp, err := artifactClient.GetObjectUploadURL(ctx, &artifactpb.GetObjectUploadURLRequest{
```

Comment:
Can you refresh my memory about when (and why) we go through `artifact-backend` and when we use the MinIO client directly? 🙏

Reply:
1. `NewMinioClientAndInitBucket` introduced by Jeremy for pipeline run log.
2. `NewMinioClient` introduced by me for pipeline output data which are blob data.

Now, there are 2 ways to upload data to blob storage.
1. Use `artifact-backend` to get the upload URL and `Put` the data with the upload URL.
2. Use minio client to upload data directly.

The reason why I did not follow the previous design is mainly that we need bucket name in the params when we `GetFile`. 
But in previous design, it assumes all data will be in the same bucket. So, I introduced `NewMinioClient`.